File: src/processors/emojis_processor.py
import json

from ..storage.database_manager import ElementsDatabaseManager


def process(file_path: str, db_path: str):
    elements_db_manager = ElementsDatabaseManager(db_path)

    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()
        data: dict[str, str] = json.loads(content)
        elements = list(data.items())
        elements_db_manager.bulk_add_elements(elements)

File: src/processors/emojis_processor.ts
import * as z from "zod";

import { ElementsDatabaseManager } from "~/db_manager";

const EmojiFileSchema = z.record(z.string(), z.string());

export async function process_emojis(file_path: string, db_path: string) {
  const elements_db_manager = new ElementsDatabaseManager(db_path);
  const file = Bun.file(file_path);
  const data = EmojiFileSchema.parse(await file.json());
  const elements = Object.entries(data);
  elements_db_manager.bulk_add_elements(elements);
}
